packages/jupyter-ai/src/components/rendermime-markdown.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useState, useEffect, useRef } from 'react';
+import ReactDOM from 'react-dom';
+
+import { CopyButton } from './copy-button';
+import { IRenderMimeRegistry } from '@jupyterlab/rendermime';
+
+const MD_MIME_TYPE = 'text/markdown';
+const RENDERMIME_MD_CLASS = 'jp-ai-rendermime-markdown';
+
+type RendermimeMarkdownProps = {
+  markdownStr: string;
+  rmRegistry: IRenderMimeRegistry;
+};
+
+function RendermimeMarkdownBase(props: RendermimeMarkdownProps): JSX.Element {
+  const [renderedContent, setRenderedContent] = useState<HTMLElement | null>(
+    null
+  );
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const renderContent = async () => {
+      const model = props.rmRegistry.createModel({
+        data: { [MD_MIME_TYPE]: props.markdownStr }
+      });
+
+      const renderer = props.rmRegistry.createRenderer(MD_MIME_TYPE);
+      await renderer.renderModel(model);
+      props.rmRegistry.latexTypesetter?.typeset(renderer.node);
+      setRenderedContent(renderer.node);
```

Comment:
Would it save one render cycle if the `setRenderedContent` was called after copy buttons are rendered? I think this would be also less error-prone with respect to interaction with react update cycle.